hubble/src/token_list.rs

```diff
@@ -0,0 +1,152 @@
+                            .find(|(chain_id, _)| *chain_id == token.0.to_string())
+                            .map(|(_, id)| {
+                                (
+                                    *id as i64,
+                                    token.1.clone(),
+                                    token.2.clone(),
+                                    token.3,
+                                    token.4.clone(),
+                                    token.5.clone(),
+                                    token.6,
+                                )
+                            })
+                    })
+                    .collect();
+
+            if filtered_tokens.is_empty() {
+                return Ok(());
+            }
+
+            // Insert or update the filtered tokens
+            crate::postgres::insert_or_update_tokens(&db, &filtered_tokens).await?;
+        }
+    }
+
+    Ok(())
+}
+
+pub async fn get_tokens(urls: TokensUrls) -> Result<Option<Vec<TokenList>>> {
+    let client = reqwest::Client::new();
+
+    let requests = urls.urls.iter().map(|url| {
+        let client = client.clone();
+
+        async move {
+            let val: serde_json::Value = client.get(url).send().await?.json().await?;
+
+            if val.get("statusCode").is_none() {
+                Ok(Some(serde_json::from_value(val).unwrap())) as Result<Option<TokenList>>
+            } else {
+                return Ok(None);
+            }
+        }
+    });
+
+    // Execute all requests simultaneously and collect the results
+    let results: Vec<Result<Option<TokenList>>> = futures::future::join_all(requests).await;
+
+    let mut tokens = Vec::new();
+    for result in results {
+        if let Ok(Some(token_list)) = result {
+            tokens.push(token_list);
+        }
+    }
+    if tokens.is_empty() {
+        Ok(None)
+    } else {
+        Ok(Some(tokens))
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TokenList {
+    pub name: String,
+    pub timestamp: String,
+    pub version: Version,
+    pub tokens: Vec<Token>,
+    #[serde(rename = "logoURI")]
+    pub logo_uri: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Version {
+    pub major: i64,
+    pub minor: i64,
+    pub patch: i64,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Token {
+    pub chain_id: i64,
+    pub address: String,
+    pub name: String,
+    pub symbol: String,
+    pub decimals: i64,
+    #[serde(rename = "logoURI")]
+    pub logo_uri: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TokenInsert {
+    pub chain_id: i64,
+    pub denom: String,
+    pub display_symbol: String,
+    pub decimals: i64,
+    pub logo_uri: Option<String>,
+    pub display_name: String,
+    pub gas_token: bool, // always false?
```

Comment:
we should insert `false` on new records and keep the existing value if we're updating. so I think you best remove the field here and update the insert statement